jdaviz/app.py

```diff
@@ -1052,13 +1052,24 @@ def _get_range_subset_bounds(self, subset_state,
                                  simplify_spectral=True, use_display_units=False):
         # TODO: Use global display units
         # units = dc[0].data.coords.spectral_axis.unit
-        viewer = self.get_viewer(self._jdaviz_helper. _default_spectrum_viewer_reference_name)
-        data = viewer.data()
-        if data and len(data) > 0 and isinstance(data[0], Spectrum1D):
-            units = data[0].spectral_axis.unit
+        if not hasattr(subset_state.att, "parent"):  # e.g., Cubeviz
+            viewer = self.get_viewer(self._jdaviz_helper._default_spectrum_viewer_reference_name)
+            data = viewer.data()
+            if data and len(data) > 0 and isinstance(data[0], Spectrum1D):
+                units = data[0].spectral_axis.unit
```

Comment:
what does `data[0]` assume?  That that is the full cube extracted spectrum?  If that is the assumption, maybe we need to be more explicit and request it by label instead?  What if that has been unloaded from the viewer?  

I still wish this could be generalized, but I see how its difficult since `data` in the else is the glue component, but `data` in the `if` is a Spectrum1D (or other translated object).  Although the else in the else then uses a handler to convert to a Spectrum1D anyways.  🤔 

If all we need here are the units of the x-axis of the spectrum viewer... can we just access the display units?  Or do we need the native units of the "reference" data?

Reply:
Fair, ignore my first line of questions then since those do apply to the old code.  But I think if we can have a general block here that applies to cubeviz and specviz2d, that would be easier to maintain than if-statements, right?  There is even a TODO comment immediately above suggesting switching to display units - so maybe we can simplify this all down to one line once @gibsongreen enables display units across all configs?

Reply:
(or maybe it makes sense to get this in like this now and either roll that into @gibsongreen's WIP or as a tech-debt follow-up after that makes it in as well?)

Reply:
ok, we have decided to revisit this after unit conversion is enabled across all configs and merge the logic as-is until then.  [:cat:]([JDAT-4822](https://jira.stsci.edu/browse/JDAT-4822))